### PR TITLE
Cart

### PR DESCRIPTION
**Pull request by AYOPELUMI**
- Created the Cart Context API
- Added the Context wrapper in the 'App.jsx' file
- Moved the 'products.json' file to the public folder
- Fixed the bugs in the Product component
- Added the 'add to cart' functionality to the productCard component
- Added Cart to the routes for 
-

### CODE REVIEW

**Comment by Divinechi** [comment]
Approved.

File: src/pages/Cart/Cart.css [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    cursor: pointer;
+  }
+  
+  .cart-container button:hover {
+    background-color: #0056b3;
+  }
+  
+  
+  
+
```
Comment (Megagig) [inline src/pages/Cart/Cart.css]: @AYOPELUMI 
@Divinechi 
@adex-badr18 

Weldone, implementing the project requirements. You all agreed to use tailwind CSS for the implementation of the styles but here you went ahead to use vanilla CSS. Please kindly redo this part of the project.

File: src/pages/Cart/Context/CartContext.jsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
+/* eslint-disable react/prop-types */
```
Comment (Megagig) [inline src/pages/Cart/Context/CartContext.jsx:1]: @AYOPELUMI 
Here you disabled the eslint from the project. please kindly do your implementation without disabling the eslint. the eslint helps us write good codes that are modular and obeys standards.

File: src/pages/Cart/Cart.jsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import React, { useContext } from 'react';
+import { CartContext } from './Context/CartContext';
+import './Cart.css'; // Import the CSS file for styling
+
```
Comment (Megagig) [inline src/pages/Cart/Cart.jsx]: @Divinechi Please kindly style the project with Tailwind as we all agreed.

**Review by Megagig — CHANGES_REQUESTED** [review]
Hi @AYOPELUMI @Divinechi 
Great job so far! 👏 You've done well, and you're almost there for the next project ⏭️, but there are a few issues 🐛 that you still need to work on.

image




Don't Worry 🙏
It's okay to have bugs in your code. 🆗 Please see this code review as an opportunity to learn practical coding skills and discover new tips that can improve your code quality 💯 and help you get even better 💼.

I will guide you on how to address these areas down below 👇.

Highlights
✅ everyone has contributed

Required Changes ♻️
Check the comments under the review.

Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Have any doubt ❓
Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

⚠️ WARNING ⚠️
Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

File: src/pages/Cart/Context/CartContext.jsx (hunk 2 above)
Comment (AYOPELUMI) [reply src/pages/Cart/Context/CartContext.jsx:1]: Thanks for the insights 
I will do that after fully pleased that all basic functionality and design layout are met

File: src/pages/Cart/Cart.jsx (hunk 3 above)
Comment (Divinechi) [reply src/pages/Cart/Cart.jsx]: Sure I have done that thank you, you can review it now.

**Comment by Megagig** [comment]
@Divinechi I am sorry we cannot merge your work into the development branch. This is because your styling has affected the initial work carried out on the project. Have a look at the screenshot of your styles.

Screenshot from 2024-06-25 07-53-26

I have disabled the ability for anyone to merge pull requests. imagine what could have happened if this pull request is merged without review. it will just revert us to square one. 

Screenshot from 2024-06-25 07-55-13

Screenshot from 2024-06-25 07-55-48

Screenshot from 2024-06-25 07-56-47

Please don't merge this pull request. 
Thanks

File: postcss.config.cjs [hunk 4]
```diff
@@ -1,6 +1,7 @@
-export default {
+module.exports = {
   plugins: {
```
Comment (Megagig) [inline postcss.config.cjs]: Why did you tamper with the Tailwind config? @Divinechi

**Review by Megagig — CHANGES_REQUESTED** [review]
@Divinechi I am sorry we cannot merge your work into the development branch. This is because your styling has affected the initial work carried out on the project. Have a look at the screenshot of your styles.

Screenshot from 2024-06-25 07-53-26

I have disabled the ability for anyone to merge pull requests. imagine what could have happened if this pull request is merged without review. it will just revert us to square one. 

Screenshot from 2024-06-25 07-55-13

Screenshot from 2024-06-25 07-55-48

Screenshot from 2024-06-25 07-56-47

Please don't merge this pull request. 
Thanks

**Comment by Megagig** [comment]
@iyke-e I saw the commit on tailwind config file which you pushed. This is to bring to your notice that the styling errors are not yet sorted out. Please kindly implement this. If you are having issues that you cannot solve, reach out to team members to help you get unstuck

**Comment by iyke-e** [comment]
Noted we're working on it

**Comment by Megagig** [comment]
@Divinechi @AYOPELUMI @iyke-e 

Weldone, on your awesome work. 

Project is approved as agreed by the reviews during the call today. 

congratulations

**Pull request merged**